### ISSUE TITLE
Result from Array is not checked

### ISSUE THREAD

**Issue by Bistard** [mdbetancourt/eslint-plugin-neverthrow#18]
I just notice that the following case is not checked by the linter:
```ts
function foo(): void {
    const arr: Result<number, string>[] = [];
    const resultFromArray = arr[0]!; // should raise an error here
}
```
